lib/platform.h

```diff
@@ -127,6 +129,8 @@ namespace cppcheck {
         }
 
         static const char *platformString(PlatformType pt) {
+            if ((pt < MIN_PLATFORM_TYPE) || (pt > MAX_PLATFORM_TYPE))
```

Comment:
For information.. I like that you moved the knowledge about the order of the enum constants. I would guess that we have a few such conditions.. the `ValueType::isIntegral()` comes to my mind.

Reply:
hmm.. I would prefer a "ASSERT" failure if there is a pt value not handled by the switch. The `default` code in the original code should have caused an assert failure.

A bit off topic.. I think it would be nice with a check that guarantee that enums do not get "wrong" values. It could be done in an addon but well I dont have time for that now :-(

Reply:
I thought about that....
"Normally", IIRC, attempting to assign an out-of-range enum value should cause an assertion.

Reply:
I had to try...
```
#include <iostream>

int main() {
    enum SeverityType
    {
        MIN_TYPE,
        none = MIN_TYPE,
        error,
        warning,
        style,
        performance,
        portability,
        information,
        debug,
        MAX_TYPE=debug
    };
    
    for(int x = static_cast<int>(MIN_TYPE); x <= static_cast<int>(MAX_TYPE); ++x)
    {
        std::cout << "Severity Type: " << static_cast<SeverityType>(x) << std::endl;
    }
    SeverityType q = static_cast<SeverityType>(99);
    std::cout << "Severity Type: " << q << std::endl;
    return 0;
}
```
So, checks are absolutely necessary.

Reply:
I added a check a few years ago in Cppcheck that would warn if mismatching enum values was assigned to an enum variable. But that proved too noisy. People apparently often do that by intention. I reverted that as far as I remember. But well, I think it would be interesting to somehow readd something like it again. Today, I would be pretty interested in a "enum verification" addon that can guarantee that enums are always "valid".

Reply:
Agreed. And this is where I'm having to go back to c++ reference docs. C++11, and later, I thought added more stringent requirements onto enums. However, 'defensive programming' seems prudent.

Reply:
I think the majority of our enums are not "enum class". For instance this `PlatformType` enum is not a "enum class". Please do not hesitate to convert to "enum class".

A few years ago we had not switched to C++11 yet. And therefore "enum class" was not allowed. But this C++11 feature is now acceptable. I ultimately want it to be possible to compile Cppcheck with ~10 years old compilers because people often use compilers for a long time.

Reply:
Does "enum class" mean that you are prevented to assign an invalid value?